staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/validator.go

```diff
@@ -43,73 +43,36 @@ type customResourceValidator struct {
-	}
 
-	customResourceObject, ok := obj.(*unstructured.Unstructured)
-	// this will never happen.
-	if !ok {
-		return field.ErrorList{field.Invalid(field.NewPath(""), customResourceObject, fmt.Sprintf("has type %T. Must be a pointer to an Unstructured type", customResourceObject))}
+	if errs := a.ValidateTypeMeta(ctx, u); len(errs) > 0 {
+		return errs
 	}
-	customResource := customResourceObject.UnstructuredContent()
 
-	if err = apiservervalidation.ValidateCustomResource(customResource, a.schemaValidator); err != nil {
-		return field.ErrorList{field.Invalid(field.NewPath(""), customResource, err.Error())}
-	}
+	var allErrs field.ErrorList
 
-	if scale != nil {
-		// validate specReplicas
-		specReplicasPath := strings.TrimPrefix(scale.SpecReplicasPath, ".") // ignore leading period
-		specReplicas, _, err := unstructured.NestedInt64(customResource, strings.Split(specReplicasPath, ".")...)
-		if err != nil {
-			return field.ErrorList{field.Invalid(field.NewPath(scale.SpecReplicasPath), specReplicas, err.Error())}
-		}
-		if specReplicas < 0 {
-			return field.ErrorList{field.Invalid(field.NewPath(scale.SpecReplicasPath), specReplicas, "should be a non-negative integer")}
-		}
-		if specReplicas > math.MaxInt32 {
-			return field.ErrorList{field.Invalid(field.NewPath(scale.SpecReplicasPath), specReplicas, fmt.Sprintf("should be less than or equal to %v", math.MaxInt32))}
-		}
-
-		// validate statusReplicas
-		statusReplicasPath := strings.TrimPrefix(scale.StatusReplicasPath, ".") // ignore leading period
-		statusReplicas, _, err := unstructured.NestedInt64(customResource, strings.Split(statusReplicasPath, ".")...)
-		if err != nil {
-			return field.ErrorList{field.Invalid(field.NewPath(scale.StatusReplicasPath), statusReplicas, err.Error())}
-		}
-		if statusReplicas < 0 {
-			return field.ErrorList{field.Invalid(field.NewPath(scale.StatusReplicasPath), statusReplicas, "should be a non-negative integer")}
-		}
-		if statusReplicas > math.MaxInt32 {
-			return field.ErrorList{field.Invalid(field.NewPath(scale.StatusReplicasPath), statusReplicas, fmt.Sprintf("should be less than or equal to %v", math.MaxInt32))}
-		}
-
-		// validate labelSelector
-		if scale.LabelSelectorPath != nil {
-			labelSelectorPath := strings.TrimPrefix(*scale.LabelSelectorPath, ".") // ignore leading period
-			labelSelector, _, err := unstructured.NestedString(customResource, strings.Split(labelSelectorPath, ".")...)
-			if err != nil {
-				return field.ErrorList{field.Invalid(field.NewPath(*scale.LabelSelectorPath), labelSelector, err.Error())}
-			}
-		}
+	allErrs = append(allErrs, validation.ValidateObjectMetaAccessor(accessor, a.namespaceScoped, validation.NameIsDNSSubdomain, field.NewPath("metadata"))...)
+	if err = apiservervalidation.ValidateCustomResource(u.UnstructuredContent(), a.schemaValidator); err != nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath(""), u.UnstructuredContent(), err.Error()))
```

Comment:
In the old code, we displayed the whole custom resource when it is invalid but I'm wondering whether we should continue doing that? IMO it does not really provide much info (though one can argue that it helps in inspection of what really went wrong) and prints out the complete thing in struct form, crowding the output.

Reply:
We don't do that anywhere else than for CRs, do we?

Reply:
> We don't do that anywhere else than for CRs, do we?

No, we don't.

Reply:
If we change this, we will do it in a follow-up.

Reply:
Just for the record: talked with @sttts on slack. We can change this as a follow-up later, if needed.